File: Node/playwright_vanilla/tests/example.spec.js
const { test, expect } = require('@playwright/test');
const AxeDevtoolsBuilder = require('@axe-devtools/playwright').default;
const AxeDevtoolsReporter = require("@axe-devtools/reporter").default;
const reporter = new AxeDevtoolsReporter('playwright', './a11y-results/homepageFlows');



test.afterAll(async ({ context }) => {
    await reporter.buildHTML("./a11y-results/html/");
    await reporter.buildJUnitXML("./a11y-results/xml/");
    await reporter.buildCSV("./a11y-results/csv/");
    await browser.close();
});


test.describe('two tests', () => {
    test('basic test_1', async ({ page, browserName }) => {
        await page.goto("https://broken-workshop.dequelabs.com/");
        const results = await new AxeDevtoolsBuilder({ page }).analyze();
        reporter.logTestResult('homepage-no-flow_' + browserName, results);
        expect(results.violations).toHaveLength(0)
    });

    test('basic test_2', async ({ page, browserName }) => {
        await page.goto("https://broken-workshop.dequelabs.com/");
        await page.locator('#main-content > div.Recipes > div:nth-child(1) > div.Recipes__card-foot > button').click();
        const results2 = await new AxeDevtoolsBuilder({ page }).analyze();
        reporter.logTestResult("homepage-altered-state_" + browserName, results2);
        expect(results2.violations).toHaveLength(0)
    });
});

File: Ruby/selenium-rspec/spec/example_spec.rb
require "spec_helper"
require 'axe/core'
require 'axe/api/run'

describe 'Analyse webpage for a11y issues' do 
  it 'Analyse broken-workshop homepage for a11y issues' do
    @driver.page.navigate.to "https://broken-workshop.dequelabs.com/"
    expect(@driver.page).to be_audited_for_accessibility.logging_results({ui_state: 'homepage-analyse'})
  end
  
  it 'Analyse broken-workshop recipe card for a11y issues' do
    @driver.page.navigate.to "https://broken-workshop.dequelabs.com/"
    @driver.page.find_element(css: '#main-content > div.Recipes > div:nth-child(1) > div.Recipes__card-foot > button').click
    expect(@driver.page).to be_audited_for_accessibility.logging_results({ui_state: 'recipe-card-analyse'})
  end

  # it 'running bash commands' do
  #   exec'echo "hi"'
  # end
    
  after(:each) do
    @driver.page.quit
  end

  after(:all) do
    logs = "/Users/dq_ankit/main_project/axe-devtools-html-api-examples/Ruby/selenium-rspec/axe-reports"
    reporter = (Dir.pwd)<<'/resources/reporter-cli-macos'
    destination = (Dir.pwd)<<'/a11y-results'
    command_html = reporter << " " << logs << " --destination " << destination << " --format html"
    puts (command_html)
    system command_html
  end

end